diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c79e47a..00f3ac7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -928,7 +928,7 @@ if (BUILD_SHARED_LIBS)
   TARGET_C_PROPERTIES (${HDF5_LIBSH_TARGET} SHARED " " " ")
   target_link_libraries (${HDF5_LIBSH_TARGET} PRIVATE ${LINK_LIBS} ${LINK_COMP_LIBS})
   if (NOT WIN32)
-    target_link_libraries (${HDF5_LIBSH_TARGET} PUBLIC ${CMAKE_DL_LIBS})
+    target_link_libraries (${HDF5_LIBSH_TARGET} PRIVATE ${CMAKE_DL_LIBS})
   endif ()
   set_global_variable (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_LIBSH_TARGET}")
   H5_SET_LIB_OPTIONS (${HDF5_LIBSH_TARGET} ${HDF5_LIB_NAME} SHARED ${HDF5_PACKAGE_SOVERSION})
@@ -943,7 +943,7 @@ if (BUILD_SHARED_LIBS)
         APPEND PROPERTY COMPILE_DEFINITIONS
             "H5_HAVE_THREADSAFE"
     )
-    target_link_libraries (${HDF5_LIBSH_TARGET} PUBLIC Threads::Threads)
+    target_link_libraries (${HDF5_LIBSH_TARGET} PRIVATE Threads::Threads)
   endif ()
 
   if (HDF5_ENABLE_DEBUG_APIS)
