diff --git a/src/libbson/CMakeLists.txt b/src/libbson/CMakeLists.txt
index 4bc2cb6..ca4d4d0 100644
--- a/src/libbson/CMakeLists.txt
+++ b/src/libbson/CMakeLists.txt
@@ -217,6 +217,10 @@ set (HEADERS_FORWARDING
    ${PROJECT_SOURCE_DIR}/src/bson/forwarding/bson.h
 )
 
+add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+
+if (NOT BSON_ENABLE_STATIC)
+
 add_library (bson_shared SHARED ${SOURCES} ${HEADERS} ${HEADERS_FORWARDING})
 set (CMAKE_CXX_VISIBILITY_PRESET hidden)
 set_target_properties (bson_shared PROPERTIES COMPILE_DEFINITIONS "BSON_COMPILATION;JSONSL_PARSE_NAN")
@@ -258,16 +262,21 @@ if (WIN32)
    # must be handled specially since we can't resolve them
    set (BSON_SYSTEM_LIBRARIES ${BSON_SYSTEM_LIBRARIES} ws2_32)
 endif ()
+endif()
 
 if (ENABLE_STATIC MATCHES "ON|AUTO")
    add_library (bson_static STATIC ${SOURCES} ${HEADERS} ${HEADERS_FORWARDING})
+   set(CMAKE_CXX_VISIBILITY_PRESET hidden)
    set_target_properties (bson_static PROPERTIES COMPILE_DEFINITIONS "BSON_COMPILATION;BSON_STATIC;JSONSL_PARSE_NAN")
    set_target_properties (bson_static PROPERTIES VERSION 0.0.0)
    set_target_properties (bson_static PROPERTIES OUTPUT_NAME "bson-static-${BSON_API_VERSION}")
+   set(THREADS_PREFER_PTHREAD_FLAG 1)
+   find_package (Threads REQUIRED)
    target_link_libraries (bson_static Threads::Threads)
    if (RT_LIBRARY)
       target_link_libraries (bson_static ${RT_LIBRARY})
    endif ()
+   find_library(M_LIBRARY m)
    if (M_LIBRARY)
       target_link_libraries (bson_static ${M_LIBRARY})
    endif ()
@@ -282,7 +291,7 @@ function (add_example bin src)
    add_executable (${bin} ${BSON_EXAMPLE_SOURCES})
 
    # Link against the shared lib like normal apps
-   target_link_libraries (${bin} bson_shared)
+   target_link_libraries(${bin} bson_shared bson_static)
 
    set (EXAMPLES ${EXAMPLES} ${bin})
 endfunction ()
@@ -303,6 +312,7 @@ endif () # ENABLE_EXAMPLES
 set (BSON_HEADER_INSTALL_DIR
    "${CMAKE_INSTALL_INCLUDEDIR}/libbson-${BSON_API_VERSION}"
 )
+if (NOT BSON_ENABLE_STATIC)
 
 install (
    TARGETS bson_shared ${EXAMPLES}
@@ -311,6 +321,7 @@ install (
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
    FRAMEWORK DESTINATION ${CMAKE_INSTALL_BINDIR} 
 )
+endif ()
 if (ENABLE_STATIC MATCHES "ON|AUTO")
    install (
       TARGETS bson_static ${EXAMPLES}
diff --git a/src/libbson/src/bson/bson-compat.h b/src/libbson/src/bson/bson-compat.h
index 68f384d..d0bc5f0 100644
--- a/src/libbson/src/bson/bson-compat.h
+++ b/src/libbson/src/bson/bson-compat.h
@@ -39,11 +39,11 @@
 
 
 #ifdef BSON_OS_WIN32
-#if defined(_WIN32_WINNT) && (_WIN32_WINNT < 0x0600)
+#if defined(_WIN32_WINNT) && (_WIN32_WINNT < 0x0602)
 #undef _WIN32_WINNT
 #endif
 #ifndef _WIN32_WINNT
-#define _WIN32_WINNT 0x0600
+#define _WIN32_WINNT 0x0602
 #endif
 #ifndef NOMINMAX
 #define NOMINMAX
